uds/uds_communications/TransportProtocols/Can/CanTp.py

```diff
@@ -395,11 +394,20 @@ def create_blockList(self, payload, blockSize):
         blockPtr = 0
 
         payloadLength = len(payload)
-        pduLength = self.__maxPduLength
-        blockLength = blockSize * pduLength
-
         working = True
         while working:
+            pduLength = (
+                next(
+                    (
+                        size
+                        for size in CAN_FD_DATA_LENGTHS
+                        if size + payloadPtr > payloadLength
+                    ),
+                    64,
+                )
+                - 1
```

Comment:
- 1 for one PCI byte ?  In a first frame PCI could be 2 bytes long..
Please check, not deep enough in those topic.
Also the lack of unittests scares me

Reply:
Added unittest